### PR TITLE
boto_asg state updates

### LINKED ISSUE #27545: boto_asg allow removing launch configuration with 'absent' state
boto_asg state should have the ability to remove the launch configuration of an ASG if the ASG is deleted by the 'absent' check.  In other words, if the boto_asg.absent state deletes an ASG, it should be able to clean up the launch config as well.

### PR DESCRIPTION
**Pull request by lomeroe**
Adding ability to delete launch config on 'absent'
correcting use of vpc_id from boto_vpc.get_subnet_association call

Resolves #27545 #27544

### CODE REVIEW

File: salt/states/boto_asg.py [hunk 1]
```diff
@@ -652,6 +653,7 @@ def _recursive_compare(v1, v2):
 def absent(
         name,
         force=False,
+        remove_lc=False,
```
Comment (jfindlay) [inline salt/states/boto_asg.py]: @lomeroe, is it possible to move this argument to the end of the list, to keep api changes to a minimum?  Thanks.

**Comment by jfindlay** [comment]
@lomeroe, again, if it is possible to split out the feature addition from the bug fix, then we can send the bug fix back to 2015.5, thanks.

**Comment by lomeroe** [comment]
@jfindlay split into two PRs, for the bug fix and the remove_lc, also updated the remove_lc in the function call order as requested...thanks for your patience with me

**Comment by jfindlay** [comment]
@lomeroe, no problem and thanks for your patience. :-)

**Comment by lomeroe** [comment]
closing in favor of split PRs #27560 and #27559

**Pull request closed without merging**